analytics/js_tracking_api.rst

```diff
@@ -0,0 +1,829 @@
+
+This class name can be changed, use ``setLinkClasses`` to define which CSS class should be tracked::
+
+    _paq.push(["setLinkClasses", className]);
+
+.. data:: className
+
+    **Required** ``string`` CSS class that should be tracked instead of ``piwik_link``
+
+Example of usage::
+
+    _paq(["setLinkClasses", "track-this-link"]);
+
+Force Tracking using JS function
+++++++++++++++++++++++++++++++++
+
+If you want to use JS to force outlink to be tracked you can add ``trackLink`` function to element ``onClick`` attribute::
+
+    _paq.push(["trackLink", linkAddress, "link", dimensions]);
+
+.. data:: linkAddress
+
+    **Required** ``string`` Address that link points to.
+
+.. data:: dimensions
+
+    **Optional** ``object`` `Custom dimension <Custom Dimensions_>`_  that should be tracked with this action. Can be multiple dimensions.
+    Written as object property using ``dimension{ID}`` notation.
+
+    Example::
+
+        {
+           dimension1: "example value",
+           dimension2: "example value"
+        }
+
+
+Example of usage
+
+.. code-block:: html
+
+    <button onClick="function(){_paq.push(['trackLink', 'http://www.example.com/example', 'link']);}">
+        Click me!
+    </button>
+
+Tracking Downloads
+``````````````````
+.. todo::
+    Is download only tracking links to files ending on extension? What about GET parameters?
+
+Default extensions recognized as download
++++++++++++++++++++++++++++++++++++++++++
+
+Following extensions are tracked as download by default:
+
+
++-------+-----+-----+-----+------+-----+-----+-----+------+-----+------+-----+---------+-----+-----+
+| 7z    | aac | arc | arj | apk  | asf | asx | avi | bin  | bz  | bz2  | csv | deb     | dmg | doc |
++-------+-----+-----+-----+------+-----+-----+-----+------+-----+------+-----+---------+-----+-----+
+| exe   | flv | gif | gz  | gzip | hqx | jar | jpg | jpeg | js  | mp2  | mp3 | mp4     | mpg | mov |
++-------+-----+-----+-----+------+-----+-----+-----+------+-----+------+-----+---------+-----+-----+
+| movie | msi | msp | odb | odf  | odg | odp | ods | odt  | ogg | ogv  | pdf | phps    | png | ppt |
++-------+-----+-----+-----+------+-----+-----+-----+------+-----+------+-----+---------+-----+-----+
+| qt    | qtm | ra  | ram | rar  | rpm | sea | sit | tar  | tbz | tbz2 | tgz | torrent | txt | wav |
++-------+-----+-----+-----+------+-----+-----+-----+------+-----+------+-----+---------+-----+-----+
+| wma   | wmv | wpd | xls | xml  | z   | zip |     |      |     |      |     |         |     |     |
++-------+-----+-----+-----+------+-----+-----+-----+------+-----+------+-----+---------+-----+-----+
+
+Adding extension to default extensions
+++++++++++++++++++++++++++++++++++++++
+
+You can add extension to default extensions list using ``addDownloadExtensions`` function::
+
+    _paq.push(["addDownloadExtensions", extensions]);
+
+.. data:: extensions
+
+    **Required** ``string`` Extensions separated by ``|`` for example ``"mhj|docx"``
```

Comment:
String of elements divided by the pipe - that's OK but this parameter also accepts array of strings as a value.